### DISCUSSION TITLE
XEP-0363: HTTP File Upload

### DISCUSSION THREAD

**Discussion by AtifShahzed** [vitalyster/SharpXMPP#116] [Ideas]
Hi,

I am using ejabberd with this client library in Xamarin Forms and pretty happy so far.
I am looking for File upload but this feature is not available in library. Can you please help me out or if any resources which are helpful so please share with me.
I have configured Rest api of ejabberd but not able to find the file upload api.
Please guide me is there any resource that help me to achieve this behavior.

Thanks

**Comment by ForNeVeR** [comment]
Usually, when you want to use any new XEP, you'll need to read that XEP, then prepare right data packets for it, and then use it.

So, which XEP are you talking about? Have you taken a look at the docs?

**Comment by LumiProj** [comment]
I am talking about 
XEP-0363
https://xmpp.org/extensions/xep-0363.html

**Comment by LumiProj** [comment]
Hi,
Any help regarding XEP-0363 using this library would be great.

**Comment by ForNeVeR** [comment]
So, it looks like you're supposed to send an IQ query to get an "upload slot", wait for the result, and then perform a HTTP request to some URL.

To send an IQ query, just use `XmppConnection.Send(iq)`, where `iq` is the `XElement` you want to send.
